src/controllers/bus.controller.ts

```diff
@@ -1,52 +1,267 @@
-import {BusesModel} from '../models/bus.model';
-import express, {Request,Response} from 'express';
-import {IBus} from '../interface/bus.interface';
-
-
-const router=express.Router();
-
-router.post('/createBus/', async(req:Request,res:Response)=>{
-    const bus=new BusesModel({
-        line_number:req.body.line_number,
-        bus_color:req.body.bus_color,
-        model:req.body.model,
-        speed:req.body.speed,
-        List_station:req.body.List_station
-    })
-    try{
-        const newBus= await bus.save();
-        res.status(201).json(newBus);
-    }catch(err){
-        res.status(400).json({message:err.message}) 
+/* eslint-disable no-plusplus */
+/* eslint-disable prefer-destructuring */
+import { Request, Response } from 'express';
+import { BusesModel } from '../models/bus.model';
+import { IBus } from '../interface/bus.interface';
+
+export const postBus = async (req: Request, res: Response) => {
+  const lineNum = await BusesModel.aggregate([
+    {
+      $project: {
+        lineNumber: '$lineNumber',
+      },
+    },
+  ]);
+  const num: number[] = [];
+  lineNum.forEach((doc) => {
+    num.push(doc.lineNumber);
+  });
+
+  const bus = new BusesModel({
+    lineNumber: req.body.lineNumber,
+    busColor: req.body.busColor,
+    model: req.body.model,
+    speed: req.body.speed,
+    stationsList: req.body.stationsList,
+  });
```

Comment:
The model should not be in the controller! remove all models from all controller files. 
That is what repositories are for.